### PR TITLE
Add body details in case of failure to get token

### PR DESCRIPTION
**Pull request by tzvatot**
We see from time to time 400 bad request returning but we do not print the error reason.
This PR should handle it by only catching the 5xx errors at first, and in case of other errors, it will be handled later in the flow.

@jhernand @nimrodshn @vkareh PTAL

### CODE REVIEW

**Comment by nimrodshn** [comment]
LGTM :100:

File: token.go [hunk 1]
```diff
@@ -276,7 +276,7 @@ func (c *Connection) sendTokenFormTimed(ctx context.Context, form url.Values) (c
 	// Read the response body:
 	body, err = ioutil.ReadAll(response.Body)
 	if err != nil {
-		err = fmt.Errorf("can't read response: %v", err)
+		err = fmt.Errorf("failed to read response: %v", err)
```
Comment (jhernand) [inline token.go]: This change doesn't belong in this patch.

File: token.go [hunk 2]
```diff
@@ -300,7 +300,7 @@ func (c *Connection) sendTokenFormTimed(ctx context.Context, form url.Values) (c
 	var msg internal.TokenResponse
 	err = json.Unmarshal(body, &msg)
 	if err != nil {
-		err = fmt.Errorf("can't parse JSON response: %v", err)
+		err = fmt.Errorf("failed to parse JSON response: %v", err)
```
Comment (jhernand) [inline token.go]: This change doesn't belong in this patch.

File: token.go [hunk 3]
```diff
@@ -285,7 +285,7 @@ func (c *Connection) sendTokenFormTimed(ctx context.Context, form url.Values) (c
 
 	// Check the response status and content type:
 	code = response.StatusCode
-	if response.StatusCode != http.StatusOK {
+	if response.StatusCode >= 500 {
```
Comment (jhernand) [inline token.go]: I think it would be better to preserve the check for StatusOK and move it to line 314; right after checking if the JSON response contains an error message. Otherwise you are assuming that 5XX responses don't have a JSON body, which may not be true.
